iotics/lib/identity/register/document_helper.py

```diff
@@ -166,3 +170,24 @@ def get_valid_issuer_key_for_auth(doc: RegisterDocument, issuer_name: str,
             IdentityValidationError: if invalid name or did
         """
         return RegisterDocumentHelper.get_valid_issuer_key_for(doc, issuer_name, get_controller_doc, include_auth=True)
+
+    @staticmethod
+    def new_random_name_for_document(doc: RegisterDocument, length: int = 10) -> str:
+        """
+        New random #name that can be added to this document
+        :param doc: existing register document
+        :param length: length of name including #
+        :return: a new random #name (not already existing in this document)
+
+        ;raises:
+            IdentityValidationError: if invalid key name (eg if length too short/long)
+        """
+        length -= 1
+        chars = string.ascii_lowercase + string.digits
+        while True:
+            new_name = ISSUER_SEPARATOR + ''.join(random.choice(chars) for _ in range(length))
+            IdentityValidation.validate_key_name(new_name)
```

Comment:
Won't this raise an error over which the caller has no control? (And would it ever raise it?)

Reply:
Ah right - so there's a max `length` - worth checking that param first instead?

Reply:
Yes, if the user specified length is too long. I could duplicate the validation to raise a specific error here. Or make the length not configurable.

Reply:
Up to you - I guess if it's internal, maybe OK to leave as-is.